kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/internal/codec/RequestDecoderTest.java

```diff
@@ -84,12 +91,63 @@ void testApiVersionsExactlyOneFrame_decoded(short apiVersion) {
                                 DecodePredicate
                                         .forFilters(FilterAndInvoker.build(
                                                 (ApiVersionsRequestFilter) (version, header, request, context) -> context.requestFilterResultBuilder()
-                                                        .forward(header, request).completed()))),
+                                                        .forward(header, request).completed())),
+                                DEFAULT_SOCKET_FRAME_MAX_SIZE_BYTES),
                         DecodedRequestFrame.class,
                         (RequestHeaderData header) -> header, true),
                 "Unexpected correlation id");
     }
 
+    @Test
+    void shouldThrowIfFirstIntGreaterThanMaxFrameSize() {
+        int socketFrameMaxSizeBytes = 10;
+        int sentMaxSizeBytes = socketFrameMaxSizeBytes + 1;
+        Assertions.assertThatThrownBy(() -> {
+            KafkaRequestDecoder kafkaRequestDecoder = getKafkaRequestDecoder(DECODE_EVERYTHING, socketFrameMaxSizeBytes);
+            byte arbitraryByte = 1; // need 5 bytes in the buffer for it to read the length
+            ByteBuf buffer = toByteBuffer(sentMaxSizeBytes, arbitraryByte);
```

Comment:
Given this is identical in all three usages would it make sense to just embed the `arbitraryByte` in `toByteBuffer` and make it `toArbitraryByteBuffer` or to `populatedByteBuffer` ?